rancid-tomatillos/src/containers/MoviePage/MoviePage.js

```diff
@@ -1,33 +1,99 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './MoviePage.scss';
 import { connect } from 'react-redux';
+import { render } from 'react-dom';
+import { fetchRatings } from '../../apiCalls';
+import { updateUserRatings } from '../../actions';
 
-const MoviePage = ({ id, movies }) => {
-  let movie = movies.movies.find(movie => movie.id === parseInt(id));
-  const { average_rating, backdrop_path, overview, poster_path, release_date, title } = movie;
-  return (
-    <main className='movie-page-main'>
-      <section className='movie-page-section' style={{backgroundImage:`url(${backdrop_path})`}} >
-        <article className='movie-page-poster' style={{backgroundImage: `url(${poster_path})`}}>
-        </article>
-          <section className='movie-info-container'>
-            <article className='movie-page-details'>
-              <h2>{title}</h2>
-              <p className='movie-overview'>{overview}</p>
-              <p>{release_date}</p>
-            </article>
-            <article className='movie-page-details'>
-              <h2>Rating</h2>
-              <h2>{average_rating}</h2>
-            </article>
-          </section>
-      </section>
-    </main>
-  )
+
+class MoviePage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      rating: null
+    }
+  }
+
+  handleRatingSelect = (e) => {
+    this.setState({ rating: e.target.value })
+  }
+
+  handleRatingSubmit = () => {
+    const { id, userId } = this.props;
+    const options = {
+      method: 'POST',
+      body: JSON.stringify({
+        movie_id: parseInt(id),
+        rating: parseInt(this.state.rating)
+      }),
+      headers: { 
+      'Content-Type': 'application/json'
+      }
+    }
+
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v1/users/${userId}/ratings`, options)
+      .then(res => res.json())
+      .then(data => {
+        fetchRatings(userId)
+          .then(data => this.props.updateUserRatings(data.ratings))
+      })
+      .catch(err => console.log(err))
+  }
+
+  render() {
+    const { id, movies, isLoggedIn } = this.props
+
+    let movie = movies.movies.find(movie => movie.id === parseInt(id));
+    const { average_rating, backdrop_path, overview, poster_path, release_date, title } = movie;
+    const rating =
+    <> 
+      <h2>Rate This Movie</h2>
```

Comment:
I think if we have time we could try and style this and make it look fancier. Maybe something like one of those wheel scroll windows or something like that: I think we could take a look at this plugin - looks pretty simple to implement and has a good 1/10 option! https://antennaio.github.io/jquery-bar-rating/

Reply:
I am game! Can you send a snippet of what it looks like or a gif?